dsgrid/config/dimensions.py

```diff
@@ -389,31 +395,81 @@ class DateTimeDimensionModel(TimeDimensionBaseModel):
         options=TimeZone.format_descriptions_for_docs(),
     )
 
-    @validator("ranges", pre=True)
-    def check_times(cls, ranges, values):
-        return _check_time_ranges(ranges, values["str_format"], values["frequency"])
+    @root_validator(pre=False)  # TODO: modify as model works with more time_type schema
+    def check_time_type(cls, values):
+        if (
+            values["class_name"] == "Time" and values["time_type"] == TimeDimensionType.DATETIME
+        ) or (
+            values["class_name"] == "AnnualTime"
+            and values["time_type"] == TimeDimensionType.ANNUAL
+        ):
+            pass
+        else:
+            raise ValueError(
+                f'time_type={values["time_type"].value} does not match class_name={values["class_name"]}. \n'
+                " * For class=Time, use time_type=datetime. \n"
+                " * For class=AnnualTime, use time_type=annual."
+            )
+        return values
 
-    @validator("frequency")
-    def check_frequency(cls, value):
-        if value == timedelta(days=366):
+    @root_validator(pre=False)
```

Comment:
Does this need to be a root_validator?

Reply:
Somehow yes, I tried it without root and it would not raise an error even though the if statement would be TRUE